library-best-practices/JLBP-10.md

```diff
@@ -0,0 +1,33 @@
+[JLBP-10] Infrequent breaking changes
```

Comment:
Do we want to phrase these as full, imperative, positive statements? If so, something like, "Maintain API stability for multiple years."

If the older version is maintained and available, is there even such a thing as a breaking change? If clients can migrate at their own pace and speed, this is not a big issue. It's more like, "here's this library that still works and here's a different library with extra features  and a new package and artifact ID you can use if you prefer."

Reply:
I like the wording of "maintain stability", but the "years" part doesn't seem to apply to all cases.

If the older version stays maintained, then you are correct, there hasn't really been a breaking change. *However*, if the older version becomes an LTS release that has a deprecation timeline, consumers will be forced to accept a breakage eventually.

Reply:
"Maintain API stability as long as needed for clients"?

It occurs to me that we don't actually care about clients that are not themselves in active development, since they'll just use the old artifacts no matter what we do. The real issue is indeed Java versions since libraries do choose to continue with older versions.

Reply:
That sounds like good wording, I'll use that.